src/sidebar/search-client.ts

```diff
@@ -94,32 +113,29 @@ export class SearchClient extends TinyEmitter {
     this._sortOrder = sortOrder;
 
     this._canceled = false;
-    /** @type {Annotation[]} */
     this._results = [];
     this._resultCount = null;
   }
 
   /**
    * Fetch a page of annotations.
    *
-   * @param {SearchQuery} query - Query params for /api/search call
-   * @param {string} [searchAfter] - Cursor value to use when paginating
+   * @param query - Query params for /api/search call
+   * @param [searchAfter] - Cursor value to use when paginating
    *   through results. Omitted for the first page. See docs for `search_after`
    *   query param for /api/search API.
-   * @param {number} [pageIndex]
    */
-  async _getPage(query, searchAfter, pageIndex = 0) {
+  async _getPage(query: SearchQuery, searchAfter?: string, pageIndex = 0) {
     const pageSize = this._getPageSize(pageIndex);
 
-    /** @type {SearchQuery} */
     const searchQuery = {
       limit: pageSize,
       sort: this._sortBy,
       order: this._sortOrder,
       _separate_replies: this._separateReplies,
 
       ...query,
-    };
+    } as SearchQuery;
```

Comment:
Done. In my head I thought of these two approaches as being much the same, since TypeScript only allows assertions to cast to a more or less specific type, but not an incompatible one. It turns out that using a type annotation instead of `as` catches mistakes such as if the literal contains "extra" property names that don't match known ones in the type.